modules/CbChargeSOMDriver/chargesom/CbChargeSOM.cpp

```diff
@@ -3,251 +3,681 @@
+            }
+
+            // forward contactor errors
+            for (i = 0; i < CB_PROTO_MAX_CONTACTORS; ++i) {
+                std::string name = "Contactor " + std::to_string(i + 1);
+
+                current_contactor_error[i] =
+                    cb_proto_contactorN_is_enabled(&tmpctx, i) && cb_proto_contactorN_has_error(&tmpctx, i);
+
+                if (current_contactor_error[i] != previous_contactor_error[i]) {
+                    EVLOG_debug << "on_contactor_error: " << i;
+                    this->on_contactor_error(name, cb_proto_contactorN_get_target_state(&tmpctx, i),
+                                             cb_proto_contactorN_is_closed(&tmpctx, i)
+                                                 ? types::cb_board_support::ContactorState::Closed
+                                                 : types::cb_board_support::ContactorState::Open);
+                }
+            }
+
+            // check for ESTOP errors
+            for (i = 0; i < CB_PROTO_MAX_ESTOPS; ++i) {
+                current_estop_tripped[i] =
+                    cb_proto_estopN_is_enabled(&tmpctx, i) && cb_proto_estopN_is_tripped(&tmpctx, i);
+                if (current_estop_tripped[i] != previous_estop_tripped[i]) {
+                    EVLOG_debug << "on_estop: " << i;
+                    this->on_estop(i, current_estop_tripped[i]);
+                    previous_estop_tripped[i] = current_estop_tripped[i];
+                }
+            }
+
+            // check for PP changes
+            current_pp_state = cb_proto_get_pp_state(&tmpctx);
+            if (current_pp_state != previous_pp_state) {
+                if (previous_pp_state != PP_STATE_MAX) {
+                    if (this->is_pluggable) {
+                        EVLOG_debug << "on_pp_change(" << previous_pp_state << " → " << current_pp_state << ")";
+                        this->on_pp_change(current_pp_state);
+                    } else {
+                        EVLOG_debug << "on_pp_change(" << previous_pp_state << " → " << current_pp_state << ")"
+                                    << " [suppressed, fixed cable]";
+                    }
+                } else {
+                    EVLOG_debug << "on_pp_change(" << previous_pp_state << " → " << current_pp_state << ")"
+                                << " [suppressed]";
+                }
+                previous_pp_state = current_pp_state;
+            }
+
+            // check for CP changes
+            current_cp_state = cb_proto_get_cp_state(&tmpctx);
+            if (current_cp_state != previous_cp_state) {
+                // the integer value representation of both enum classes are the same but we have to cast over int
+                unsigned int cp_state_as_number = static_cast<unsigned int>(current_cp_state);
+
+                types::cb_board_support::CPState new_cp_state =
+                    static_cast<types::cb_board_support::CPState>(cp_state_as_number);
```

Comment:
Can we not cast directly?
```suggestion
                types::cb_board_support::CPState new_cp_state =
                    static_cast<types::cb_board_support::CPState>(current_cp_state);
```